test-app/app/src/main/java/com/tns/NativeScriptSyncService.java

```diff
@@ -268,16 +268,19 @@ private void deleteRemovedFiles(File sourceDir, String sourceRootAbsolutePath, S
         if (files != null) {
             for (int i = 0; i < files.length; i++) {
                 File file = files[i];
+                String targetFilePath = file.getAbsolutePath().replace(sourceRootAbsolutePath, targetRootAbsolutePath);
+                File targetFile = new File(targetFilePath);
                 if (file.isFile()) {
                     if (logger.isEnabled()) {
                         logger.write("Syncing removed file: " + file.getAbsolutePath().toString());
                     }
 
-                    String targetFilePath = file.getAbsolutePath().replace(sourceRootAbsolutePath, targetRootAbsolutePath);
-                    File targetFile = new File(targetFilePath);
                     targetFile.delete();
                 } else {
                     deleteRemovedFiles(file, sourceRootAbsolutePath, targetRootAbsolutePath);
+                    if (targetFile.listFiles().length == 0) {
```

Comment:
It cleans the empty folders after it's done deleting files.
It's not necessary, I just did it so we don't pollute the device with empty folders. Plus to keep behavior the same as on iOS devices.